src/main/java/com/artipie/cache/CachedCreds.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.cache;
+
+import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlInput;
+import com.amihaiemil.eoyaml.YamlMapping;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import com.artipie.asto.ext.PublisherAs;
+import com.artipie.asto.misc.UncheckedIOFunc;
+import com.artipie.misc.ArtipieProperties;
+import com.artipie.misc.Property;
+import com.artipie.repo.ConfigFile;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementation of cache for credentials with similar configurations
+ * in Artipie settings using {@link LoadingCache}.
+ * @since 0.23
+ */
+public final class CachedCreds implements CredsConfigCache {
+    /**
+     * Cache for storages settings.
+     */
+    private static LoadingCache<Metadata, CompletionStage<YamlMapping>> creds;
+
+    static {
+        CachedCreds.creds = CacheBuilder.newBuilder()
```

Comment:
I'm not a fun of static section, for example, it does not allow to properly test the class: if `LoadingCache` would be normal private field and corresponding ctor, `invalidateAll` could be checked very easy... Don't you doncider changing approach?

Reply:
It sounds reasonable because now an instance of class is created on the top level only once (e.g. the similar way of initialization of a static field). And it is made in ctor of `SettingsCaches.All`